eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/tcp/client/recommend/EventMeshRecommendImpl.java

```diff
@@ -226,15 +228,16 @@ private String recommendProxy(final Map<String, String> eventMeshMap,
             log.info("eventMeshMap:{},clientDistributionMap:{},group:{}", eventMeshMap, clientDistributionMap, group);
         }
 
-        if (!eventMeshMap.keySet().containsAll(clientDistributionMap.keySet())) {
+        Set<String> clientdDistributionSet=new HashSet<>(clientDistributionMap.keySet());
+        clientdDistributionSet.retainAll(eventMeshMap.keySet());
```

Comment:
Retains only the elements in this collection that are contained in the specified collection (optional operation). In other words, removes from this collection all of its elements that are not contained in the specified collection.
This implementation iterates over this collection, checking each element returned by the iterator in turn to see if it's contained in the specified collection. If it's not so contained, it's removed from this collection with the iterator's remove method.
I think Iteration over keyset can not be avoided .

Reply:
@jonyangx  Idea was to use retainAll() method to get intersection of two sets of keys, if there exists one element in intersection set which indirectly proves there exist proxy(key) not register(i.e. not present in eventMeshMap) but exist in distributionMap"

Reply:
public boolean removeAll(Collection<?> c) {
        Objects.requireNonNull(c);
        boolean modified = false;
        Iterator<?> it = iterator();
        while (it.hasNext()) {
            if (c.contains(it.next())) {
                it.remove();
                modified = true;
            }
        }
        return modified;
    }

The removeAll method also uses iterator()  , and use remove operation.  Why not use containAll directly?

Reply:
Ok. Got it. Then would close the PR and issue.